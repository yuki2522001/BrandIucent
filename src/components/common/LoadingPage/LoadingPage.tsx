import React, { memo } from "react";
import "./loadingPage.css";

const LoadingPage: React.FC = memo(({}) => {
  return (
    <div data-testid="loading-page" className="loader" />
  );
});

export default LoadingPage;
