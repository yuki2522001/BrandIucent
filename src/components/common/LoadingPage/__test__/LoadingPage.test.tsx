import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import LoadingPage from "../LoadingPage";

describe("LoadingPage component", () => {
  test("should render LoadingPage component", () => {
    const { getByTestId } = render(<LoadingPage />);
    expect(getByTestId("loading-page")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<LoadingPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
