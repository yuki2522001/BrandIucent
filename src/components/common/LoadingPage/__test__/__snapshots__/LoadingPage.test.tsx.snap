// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingPage component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="loader"
    data-testid="loading-page"
  />
</DocumentFragment>
`;
